File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private Integer maxHeight;

    public int maxDepth(TreeNode root) {
        maxHeight = 0;
        max(root, 0);
        return maxHeight;
    }

    public void max(TreeNode root, int currHeight) {
        if(root == null) {
            maxHeight = Math.max(currHeight, maxHeight);
            return;
        }

        max(root.left, currHeight+1);
        max(root.right, currHeight + 1);
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        p1 = 0
        p2 = p1

        maxNum = 0

        while p1 < len(arr) - 1:
            p2 = p1 + 1
            maxNum = 0
            while p2 < len(arr):
                maxNum = max(maxNum, arr[p2])
                p2+=1
            arr[p1] = maxNum
            p1+=1

        arr[len(arr) - 1] = - 1

        return arr